e2e/fixtures/create-pages/src/entries.tsx

```diff
@@ -99,6 +100,44 @@ const pages: ReturnType<typeof createPages> = createPages(
       path: '/404',
       component: () => <h2>Not Found</h2>,
     }),
+
+    createApi({
+      path: '/api/hi.txt',
+      mode: 'static',
+      method: 'GET',
+      handler: async () => {
+        const hiTxt = await readFile('./private/hi.txt');
+        return new Response(hiTxt);
+      },
+    }),
+
+    createApi({
+      path: '/api/hi',
+      mode: 'dynamic',
+      method: 'GET',
+      handler: async () => {
+        return new Response('hello world!');
+      },
+    }),
+
+    createApi({
+      path: '/api/hi',
+      mode: 'dynamic',
+      method: 'POST',
+      handler: async (req) => {
+        const body = await req.text();
+        return new Response(`POST to hello world! ${body}`);
+      },
+    }),
+
+    createApi({
+      path: '/api/empty',
+      mode: 'static',
```

Comment:
It's probably a doc issue, but if we use static mode, we recommend using file extension like `/api/empty.txt`. Otherwise, the static build will be `/api/empty/index.html`.

Reply:
We *can* enforce that with types, but I think docs are the right way to encourage that to start.